src/pages/Share/ShareTab.tsx

```diff
@@ -1,9 +1,122 @@
-import React from 'react';
-import ShareTabParticipantsSelector from '@components/Share/ShareTabParticipantsSelector';
+import React, {useEffect, useMemo} from 'react';
+import {useOnyx} from 'react-native-onyx';
+import {useOptionsList} from '@components/OptionListContextProvider';
+import SelectionList from '@components/SelectionList';
+import InviteMemberListItem from '@components/SelectionList/InviteMemberListItem';
+import useDebouncedState from '@hooks/useDebouncedState';
+import useFastSearchFromOptions from '@hooks/useFastSearchFromOptions';
+import useLocalize from '@hooks/useLocalize';
+import useNetwork from '@hooks/useNetwork';
+import useScreenWrapperTranstionStatus from '@hooks/useScreenWrapperTransitionStatus';
+import useThemeStyles from '@hooks/useThemeStyles';
+import {getOptimisticChatReport, saveReportDraft, searchInServer} from '@libs/actions/Report';
+import {saveUnknownUserDetails} from '@libs/actions/Share';
+import Navigation from '@libs/Navigation/Navigation';
+import {combineOrderingOfReportsAndPersonalDetails, getHeaderMessage, getSearchOptions} from '@libs/OptionsListUtils';
+import type {OptionData} from '@libs/ReportUtils';
+import StringUtils from '@libs/StringUtils';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
 
+const defaultListOptions = {
+    userToInvite: null,
+    recentReports: [],
+    personalDetails: [],
+    currentUserOption: null,
+    categoryOptions: [],
+};
+
 function ShareTab() {
-    return <ShareTabParticipantsSelector detailsPageRouteObject={ROUTES.SHARE_DETAILS} />;
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const {isOffline} = useNetwork();
+    const [textInputValue, debouncedTextInputValue, setTextInputValue] = useDebouncedState('');
+    const [betas] = useOnyx(ONYXKEYS.BETAS);
+
+    const {options, areOptionsInitialized} = useOptionsList();
+    const {didScreenTransitionEnd} = useScreenWrapperTranstionStatus();
+    const [isSearchingForReports] = useOnyx(ONYXKEYS.IS_SEARCHING_FOR_REPORTS, {initWithStoredValues: false});
+
+    const offlineMessage: string = isOffline ? `${translate('common.youAppearToBeOffline')} ${translate('search.resultsAreLimited')}` : '';
+    const showLoadingPlaceholder = useMemo(() => !areOptionsInitialized || !didScreenTransitionEnd, [areOptionsInitialized, didScreenTransitionEnd]);
+
+    const searchOptions = useMemo(() => {
+        if (!areOptionsInitialized) {
+            return defaultListOptions;
+        }
+        return getSearchOptions(options, betas ?? [], false);
```

Comment:
As we are not filtering out the read-only reports, it caused an issue https://github.com/Expensify/App/issues/62199